app/controllers/posts_controller.rb

```diff
@@ -36,6 +36,11 @@ def show
     @post = Post.where(author: params[:user_id]).find(params[:id])
   end
 
+  def delete
+    Post.find(params[:post_id]).destroy
+    redirect_to user_posts_path(params[:user_id])
+  end
+
   private
 
   def post_params

```

Comment:
- [ ] From the requirements of this project, Post authorization equally uses `CanCanCan`. Hence your Post Controller should contain the line ` load_and_authorize_resource` and should look like something of this sort:

```ruby
  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```



Thanks :pray: